drv/lpc55-swd/build.rs

```diff
@@ -36,12 +38,41 @@ fn generate_swd_functions(config: &TaskConfig) -> Result<()> {
     let dest_path = out_dir.join("swd.rs");
     let mut file = std::fs::File::create(dest_path)?;
 
-    let out_cfg = &config.out_cfg;
-    let in_cfg = &config.in_cfg;
     let spi_periph = quote::format_ident!("Fc{}", config.spi_num);
     let flexcomm = quote::format_ident!("FLEXCOMM{}", config.spi_num);
     let spi_regs = quote::format_ident!("SPI{}", config.spi_num);
 
+    let gpio_config_fn = |pins: &[PinConfig], name| {
```

Comment:
This doesn't look like it's closing over any other locals, in which case I'd suggest rewriting it as a `fn` rather than a closure. Among other things, debugger feedback is clearer (it'll have a name).

If I've missed the part where it's closing over state, ignore me.

Reply:
It was closing over "I don't want to import `proc_macro2` to give an explicit type to `name`" 😛 

Moving the logic into a loop cleans this up, done in f7eb5e07dd77